openedx/openedx-learning
# Adds Taxonomy, Tag, ObjectTag models and APIs

## Description

Implements the Taxonomy, Tag, and ObjectTag models, APIs, and permissions described in the [discovery document](https://docs.google.com/document/d/13zfsGDfomSTCp_G-_ncevQHAb4Y4UW0d_6N8R2PdHlA/edit?usp=sharing) for the tagging app.

## Supporting information

Relates to https://github.com/openedx/modular-learning/issues/63

## Testing instructions

Manual testing is possible but not very useful, since we don't yet have views for these models and APIs.

* Ensure that all models and methods from the discovery doc are implemented.
* Ensure that the tests cover the expected behavior of the models, APIs and permissions as described in the discovery doc.

## Other information

The `import_tags`, `export_tags` and `autocomplete_tags` will be added by a subsequents PRs

SystemTaxonomy will be added by a subsequent PR as a subclass of Taxonomy.

ContentTaxonomy and ContentTag will subclass these models, and will be implemented in edx-platform `openedx.features.content_tagging`.

## To do before merge

- [x] use multi-collation field types added by https://github.com/openedx/openedx-learning/pull/51
- [x] squash migrations into a single `0001_initial.py`
- [ ] squash and merge this change (don't just merge)
